### PR TITLE
docker-java version updated to 3.1.5

### PR DESCRIPTION
**Pull request by NehaNaithani**

### CODE REVIEW

File: iofog-agent-client/src/main/java/org/eclipse/iofog/Client.java [hunk 1]
```diff
@@ -200,6 +200,8 @@ private static String showHelp() {
                 "                 -cf <#seconds>          Set the get changes frequency\n" +
                 "                 -df <#seconds>          Set the post diagnostics frequency\n" +
                 "                 -sd <#seconds>          Set the scan devices frequency\n" +
+                "                 -pf <#seconds>          Set the docker pruning frequency\\n" +
+                "                 -dt <#percentage>          Set the available disk threshold\\n" +
```
Comment (baghbidi) [inline iofog-agent-client/src/main/java/org/eclipse/iofog/Client.java]: Extra spaces

File: iofog-agent-daemon/src/main/java/org/eclipse/iofog/command_line/CommandLineAction.java [hunk 2]
```diff
@@ -353,6 +371,8 @@ private static String showHelp() {
 			"                 -cf <#seconds>          Set the get changes frequency\\n" +
 			"                 -df <#seconds>          Set the post diagnostics frequency\\n" +
 			"                 -sd <#seconds>          Set the scan devices frequency\\n" +
+			"                 -pf <#seconds>          Set the docker pruning frequency\\n" +
+			"                 -dt <#percentage>          Set the available disk threshold\\n" +
```
Comment (baghbidi) [inline iofog-agent-daemon/src/main/java/org/eclipse/iofog/command_line/CommandLineAction.java]: Extra spaces

File: iofog-agent-daemon/src/main/java/org/eclipse/iofog/command_line/CommandLineConfigParam.java [hunk 3]
```diff
@@ -52,7 +52,9 @@
     GPS_COORDINATES ("", "", "gps_coordinates", "gpscoordinates"),
     POST_DIAGNOSTICS_FREQ ("10", "df", "post_diagnostics_freq", "postdiagnosticsfreq"),
     FOG_TYPE ("auto", "ft", "fog_type", ""),
-    DEV_MODE ("on", "dev", "dev_mode", "");
+    DEV_MODE ("on", "dev", "dev_mode", ""),
+    DOCKER_PRUNING_FREQUENCY ("2000", "pf", "docker_pruning_freq", "dockerPruningFrequency"),
```
Comment (baghbidi) [inline iofog-agent-daemon/src/main/java/org/eclipse/iofog/command_line/CommandLineConfigParam.java]: The value should be in seconds, to be consistent with other config values like checkStatusFrequency,...

File: iofog-agent-daemon/src/main/java/org/eclipse/iofog/field_agent/FieldAgent.java [hunk 4]
```diff
@@ -324,27 +325,30 @@ private void verificationFailed(Exception e) {
                 StatusReporter.setFieldAgentStatus().setLastCommandTime(lastGetChangesList);
 
                 JsonObject changes = result;
-                if (changes.getBoolean("deleteNode") && !initialization) {
+                if (changes.containsKey("deleteNode") && (changes.getBoolean("deleteNode") && !initialization)) {
```
Comment (baghbidi) [inline iofog-agent-daemon/src/main/java/org/eclipse/iofog/field_agent/FieldAgent.java]: You could use defaultValue here IMO:
```
if (changes.getBoolean("deleteNode", false) && !initialization)
```

**Review by baghbidi — CHANGES_REQUESTED** [review]

File: iofog-agent-daemon/src/main/java/org/eclipse/iofog/field_agent/FieldAgent.java (hunk 4 above)
Comment (NehaNaithani) [reply iofog-agent-daemon/src/main/java/org/eclipse/iofog/field_agent/FieldAgent.java]: updated

File: iofog-agent-daemon/src/main/java/org/eclipse/iofog/command_line/CommandLineConfigParam.java (hunk 3 above)
Comment (NehaNaithani) [reply iofog-agent-daemon/src/main/java/org/eclipse/iofog/command_line/CommandLineConfigParam.java]: updated to 60

File: iofog-agent-daemon/src/main/java/org/eclipse/iofog/command_line/CommandLineAction.java (hunk 2 above)
Comment (NehaNaithani) [reply iofog-agent-daemon/src/main/java/org/eclipse/iofog/command_line/CommandLineAction.java]: removed

File: iofog-agent-client/src/main/java/org/eclipse/iofog/Client.java (hunk 1 above)
Comment (NehaNaithani) [reply iofog-agent-client/src/main/java/org/eclipse/iofog/Client.java]: removed

**Review by baghbidi — APPROVED** [review]

**Pull request merged**